### PR TITLE
Bugfix/504 rdb rating reader

### LINKED ISSUE #504: RdbReader adds extra 0.0 -> 0.0 element
**Describe the bug**
The RDB reader inserts a data element that doesn't exist

**To Reproduce**
1. Get an RDB file that has stages below zero such as: [waterdata.usgs.gov/nwisweb/data/ratings/exsa/USGS.14365400.exsa.rdb](https://waterdata.usgs.gov/nwisweb/data/ratings/exsa/USGS.14365400.exsa.rdb)
2. Run a RdbRating computation with a timeseries linked to this rating that has values at 0.00
3. See that input values of 0 ft result in 0 cfs instead of (at time of report ~28 cfs

**Expected behavior**
No additional values should be added to the ratings list.

**Screenshots**

**Desktop/ServerSide (please complete the following information):**
 - Version 6.8+

**Additional context**
Removing any blank lines in the file, especially the data area, will allow the table to be loaded correctly.

### PR DESCRIPTION
**Pull request by MikeNeilson**
## Problem Description

Fixes #504. 

Describe the problem you are trying to solve.

## Solution

Prevent over processing that results in extra elements instead of an exception.

## how you tested the change

Manually running the computation. Will write test later, just getting this open for initial review.
Unit tests of the RdbRating code.

## Where the following done:

- [ ] Tests. Check all that apply:
   - [x] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [ ] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

**Comment by MikeNeilson** [comment]
I moved some of the nested classes out to their own files. I'm not sure why but I keep getting class not found errors and I'm pretty sure it's just language changes.

File: src/main/java/decodes/tsdb/compedit/CompParmTableModel.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+import javax.swing.table.AbstractTableModel;
+
+import decodes.db.Constants;
+import decodes.db.DataType;
+import decodes.db.SiteName;
+import decodes.gui.SortingListTableModel;
+import decodes.tsdb.DbCompParm;
+import decodes.tsdb.DbComputation;
+import decodes.tsdb.TimeSeriesDb;
+
```
Comment (ktarbet) [inline src/main/java/decodes/tsdb/compedit/CompParmTableModel.java:17]:
```suggestion
/**
 * A table model used for displaying time-series computation parameters in the Computation Editor user interface.
 */
```

**Review by ktarbet — CHANGES_REQUESTED** [review]
It is great to see more tests!

PR Looks good, I have some minor suggestions.

File: src/main/java/decodes/comp/LookupTable.java [hunk 2]
```diff
@@ -64,312 +11,345 @@
...
+        {
+            return 0.0;
+        }
+        RatingPoint rp = points.get(0);
+        return rp.indep;
+    }
+
+    public int size() { return points.size(); }
+
+    public double getMaxInput()
```
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply src/main/java/decodes/comp/LookupTable.java:305]: Sames as above comment.

**Review by ktarbet — APPROVED** [review]

**Pull request merged**